### ISSUE TITLE
GlobalsToArgumentsTrans fails to spot symbols declared in outer module scope

### ISSUE THREAD

**Issue by arporter** [stfc/PSyclone#649]
In #615 we are adding support for unqualified USE statements and that has shown-up some problems with (at least) the GlobalsToArguments transformation. In particular, it fails to spot cases where a kernel accesses module data that is declared in the outer, enclosing module scope. It's not obvious how to fix this since the symbol currently does not appear in the symbol table of the kernel - only in that of the enclosing Container. We therefore have no way of knowing that the kernel accesses such a symbol without walking through the PSyIR of the kernel body.

**Comment by rupertford** [comment]
Yes, this is a general problem at the moment e.g. when looking to see if a variable name has already been declared. I was thinking of fixing this by having a symbol table check itself and all parent symbol tables by default when looking for a symbol (with an option to only look locally), but there may be better solutions.
